### ISSUE TITLE
Build clean data set of jackpots and probabilities by lottery

### ISSUE THREAD

**Issue by bblockwood** [huntallcott/Lotteries#23]
The deliverable for this issue is a clean Stata data set of prize-and-probability pairs at the state X lottery X week level.

**Comment by jackblun** [comment]
@anjoung thats a great graph and shows that the data from La Fleurs and your gathered data roughly align in the way we would expect. Great job! One issue is time trends. Since these are all in nominal terms, it would be great to see the same thing with year fixed effects to take out some of this.

Regarding the jackpot annuity issue. My recollection is that it is fairly straightforward to related the lump-sum jackpot to the annuity equivalent. Some info [here](https://www.annuity.org/annuities/types/lottery/). Its a huge difference so we do need to deal with it.

The way you have assigned sales to jackpot draws seems fine for now to check broad patterns in the data, but we may need to do something different. If thats the easiest data structure to work with, then you can stick with that.

If possible, would be good to include powerplay info too.

Thanks again!

**Comment by jackblun** [comment]
Thanks for the update and the detailed research. We might have to do some work later on the annuity vs cash issue, since it does make a big difference. Could you see how much variation there actually is? You should for example be able to find out what the advertised jackpot was for the most recent Powerball and mega millions winnings, then calculate the percentage difference to what you see in the data. If its always say approximately 40%, with some minor differences over time, that would be useful to know.

On the MM PB combined binned scatter plots, have you tried generating a control indicating whether the data is mega millions or Powerball? If you look at the number for sales, the Powerball sales figures are far lower than those for mega millions, so its kind of weird to pool them. If you generate an indicator for whether each datapoint is mega millions or Powerball, then use absorb(), you should get the positive slope back. I may be wrong but worth checking.

Your next steps suggestions sound right.

**Comment by anjoung** [comment]
Will look into most recent winnings soon, but you were right about using absorb().

![overal](https://user-images.githubusercontent.com/30239050/52440385-01103a80-2aec-11e9-99fc-b5e043d162a6.png)

**Comment by anjoung** [comment]
@jackblun Just got a response from Wisconsin and here is what they said
> I spoke with our web team and the numbers on our site use the original estimated jackpot amounts for that draw, while the Powerball site run by MUSL likely is updated with the final jackpot amount. While our website states "estimated jackpot" in other pages, I've asked them to add that point to that webpage. Thanks for pointing that out!

Since they use the advertised jackpot amount, should I switch over to the Wisconsin data as our main data source for jackpot earnings? I am still waiting to get a response on what exactly the PB website data is measuring.

**Comment by jackblun** [comment]
Brilliant. Sounds then like we might be better off using Wisconsin's data, given we are interested in how demand responds to the advertised jackpot. Could you produce something comparing the two?

**Comment by anjoung** [comment]
Do you mean comparing jackpot amounts from Wisconsin to sales, or comparing Powerball's website data with Wisconsin's data?

**Comment by jackblun** [comment]
Sorry, to clarify. I meant comparing powerball to wisconsin so that we can see where there are differences.

**Comment by anjoung** [comment]
I calculate difference = jackpot_wisc - jackpot_pb.

First, here are the percentage of observations for which we have differences:
**Overall**: 67.5%
**Among Winners**: 98.8%
**Among Non-Winners**: 63.7%
**Among Cash Claims**: 100%
**Among Annuity Claims**: 78.6%

I also generated scatter plots to see if differences change over time and I saw nothing on that front (mapped out both the difference and an indicator for whether or not there was a difference). 
**Differences**
![diffs](https://user-images.githubusercontent.com/30239050/52495316-5147d500-2b9e-11e9-9901-ed4ddd87c544.png)

**Indicator**
![diffs_indic](https://user-images.githubusercontent.com/30239050/52495317-5147d500-2b9e-11e9-8d4b-8ab3e6650ed2.png)


Here are some histograms. The y-axis is density and x-axis is just difference = jackpot_wisc - jackpot_pb

**Overall**: 
![overall missing](https://user-images.githubusercontent.com/30239050/52495146-d8e11400-2b9d-11e9-8a6c-7de1f3e8f9ec.png)

**Among Winners**: 
![winners missing](https://user-images.githubusercontent.com/30239050/52495171-e6969980-2b9d-11e9-975d-fb6b1f4220bc.png)

**Among Non-Winners**: 
![non-winners missing](https://user-images.githubusercontent.com/30239050/52495170-e6969980-2b9d-11e9-958f-a33e9f112ab6.png)

**Among Cash Claims Winners**: 
![cash claim missing](https://user-images.githubusercontent.com/30239050/52495169-e6969980-2b9d-11e9-9baf-dc67814af25b.png)

**Among Annuity Claims Winners**:
![annuity claim missing](https://user-images.githubusercontent.com/30239050/52495168-e6969980-2b9d-11e9-9fcd-0fe6f53f2404.png)

**Comment by jackblun** [comment]
Okay, thank you. My summary of this is that those that take lump sums tend to get between 50% and 70% (pre tax) of the advertised annuity value. This is consistent with what we know from previous research and shows the data is sensible. Given this, I stick to my original suggestion of using the Wisconsin data on advertised jackpots over the directly-provided powerball data.

**Comment by anjoung** [comment]
Just pushed code to make the Wisconsin data the main jackpot variable.

**Comment by bblockwood** [comment]
@anjoung picking this task back up, as I look at _DataStorageCodebooks/Data Overview.xlsx_ it appears that we have 12 different lotteries across 7 states for which jackpot data can be merged to LaFleur's. Can you please carry out the following next steps?
1. Clean up /DataStorageCodebooks/Draw Game Data/ so that it’s clear where the data is located (e.g., one subfolder for each state, with consistent naming conventions). 
2. Write a merge script that exports a cleaned data set for each of these lotteries, using the same state-game-key naming convention developed for the LaFleur data. In doing so:
    1. Please start by confirming that we have LaFleur sales data for each of these 12 — there's no reason to clean jackpot data for a lotto without sales data.
    2. Then prioritize the high "bang-for-the-buck" lotteries, in terms of your time. For example, the TX data seems little cumbersome, parsing all the text with the CSV subfiles, whereas FL seems straightforward. 

Thanks!

**Comment by anjoung** [comment]
Update! I finished up merging the sales and jackpot data for Texas Lotto. I am going to spend tomorrow morning just checking it (running summ stats and drawing up some figures), but if everything looks good, my plan is to start working on Arizona and Louisiana next where I believe I might also have zip code level sales data from the state. 

(Say believe because it looks like zip code level data, but haven't had a chance to dig into the data and see if things are exactly as they seem.)

**Comment by anjoung** [comment]
Update: I ended up cleaning North Carolina's prize data, but realized we don't have sales data for this state. Realized that I didn't include in my memo any details about which states for which we have no data. I will add this note at the top of the memo on Monday.

Wrapped up Texas and started and finished Ohio. For Ohio, there were two lottery games.

**Comment by bblockwood** [comment]
Thanks @anjoung. Please do prioritize the lotteries that give the greatest return on your time, in terms of getting a merged jackpot+sales data set. So definitely depriotize any lottos for which we don't have sales data. (By lottery name, not just by state.) But also any for which it seems like the merge itself will be especially time consuming. 

Thanks for your work on this.

**Comment by anjoung** [comment]
Merged Florida (FL Lotto, Mega Money, Lucky Money) and Illinois (IL Lotto). Need to wrap up formatting stuff, but that will take a few minutes. 

Reaching out to FL Lottery commission to get some clarification on Mega Money and Lucky Money regarding prize structure. Also, in working on this, I uncovered that Mega Money was not a jackpot game until January 30th, 2004. However, we have sales data for all of January 2004. I flagged this data as poor quality in our main data, and also added it to the memo.

**Comment by anjoung** [comment]
Wrapped up FL and IL data. Started working on MA, but ran into data issues; sent an email to lottery commission. Switched and started working on PA, which has a lot of formatting issues. Should have that sorted tomorrow, after which I'll have a better sense of the data that I am working with.

Also received a response from AZ regarding my data request. Will check it out tomorrow to see if we have everything we need.

**Comment by anjoung** [comment]
Merged data for PA. Starting to dig into some of the more challenging states. Will check out AZ tomorrow, and if that's not fruitful start trying to scrape data for CT.

**Comment by anjoung** [comment]
Started cleaning WA but realized some data was missing. Sent a data request in to WA Lotto Commission.

Got my webscraping script to run and I collected CT small prize data. Working on cleaning and merging it.

**Comment by anjoung** [comment]
Giving an update on where things stand. I have merged together data from PB, MM, AZ, CT, FL, LA, MA, OH, TX, and WA. I just received the data from AZ, so spent some of my afternoon adding it. I have data from IL and PA that could easily be added, but ran into some problems. 

For IL lotto, I suspect that the prize data is incorrectly reported by the lotto commission. My initial reason for concern was that the EV for draws was huge. I looked at the raw data and saw that up to 6/27/2012 they were giving out tens of millions in their jackpot prize, with a starting jackpot of $20,000,000. However, my understanding was that the jackpot started at $2,000,000 which would bring the EV of draws to a more reasonable level.

For PA, I have complete data for a game called Match 6. I reviewed the game structure again and realized that the payout structure is really complicated, and am unsure how to calculate the EV for this game. At the very least, I need to compile a list of questions for the PA lottery. 

Also, WA has a really interesting feature where the EV is greater than the ticket price. It's actually currently like that. On May 21st (today), [WA Lotto](https://www.walottery.com/JackpotGames/Lotto.aspx) has a jackpot value of $9.3 million and the likelihood of winning the jackpot is 1 out of 6.99 million.
